include/zephyr/drivers/retention.h

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API for data retention drivers
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_RETENTION_
+#define ZEPHYR_INCLUDE_DRIVERS_RETENTION_
+
+#include <stdint.h>
+#include <stddef.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Data Retention Driver Interface
+ * @defgroup retention_interface Data Retention Driver Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+/**
+ * @typedef	retention_details_api
+ * @brief	
+ * See auxdisplay_custom_command() for argument description
+ */
+//typedef int32_t (*retention_details_api)(const struct device *dev, ); //size of area, start offset, etc.
+
+/**
+ * @typedef	retention_is_valid_api
+ * @brief	Callback API to check if retained data is valid (correct header and checksum,
+ *		if enabled).
+ * See auxdisplay_custom_command() for argument description
+ */
+typedef int32_t (*retention_is_valid_api)(const struct device *dev, bool *valid);
```

Comment:
This function could potentially return 1 of data is valid, 0 if it is not, and < 0 if error.

Reply:
Updated